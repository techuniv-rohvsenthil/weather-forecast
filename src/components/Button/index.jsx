import React from 'react';
import propTypes from 'prop-types';
import * as styles from './index.module.css';


const Button = (props) => {
  const { text, testID } = props;
  return (
    <div>
      <button className={styles.Button} data-testid={testID} type="button">
        {text}
      </button>
    </div>
  );
};

Button.propTypes = {
  testID: propTypes.string.isRequired,
  text: propTypes.string.isRequired,
};

export default Button;
