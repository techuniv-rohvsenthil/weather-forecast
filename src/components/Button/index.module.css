.Button{
    background-color: white;
    height: 2rem;
    width: 10em;
    color: steelblue;
    font-size: 1.1rem;
    border-radius: 12px;
    margin: 1rem;
}