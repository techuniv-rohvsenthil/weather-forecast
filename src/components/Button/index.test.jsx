import React from 'react';
import { render} from '@testing-library/react';
import Button from './index';


describe('the button component', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<Button testID="test-btn">Test text</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
