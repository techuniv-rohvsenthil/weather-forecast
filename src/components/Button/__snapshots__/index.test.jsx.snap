// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the button component should render correctly 1`] = `
<DocumentFragment>
  <div>
    <button
      data-testid="test-btn"
      type="button"
    />
  </div>
</DocumentFragment>
`;
